### ISSUE TITLE
Using Polls at scale

### ISSUE THREAD

**Issue by burnoutberni** [nextcloud/polls#1450]
As I've mentioned in another issue, my organization was planning to use this app to conduct a bunch of votes at our online General Assembly last weekend and I'd like to share a few words on this matter, maybe they end up being useful for somebody here.

We're on a [managed Nextcloud](https://www.hetzner.com/storage/storage-share) and had around 120 participants using the Polls app simultaneously, so we had both very little means to influence the smooth functioning of the system (e.g. no way to upgrade resources), but at the same time also a rather high load. Indeed, on first try, the cloud immediately crashed and went down for a few minutes, so for a while we ended up using a completely different system that we had prepared as a plan B.

When debugging, I found out that the combination of the Polls and the User Status app resulted in a lot of requests (`number of participants²`). That might not usually be a problem, if the participants log in and partake in the poll over a longer period of time, but since all of them responded within a 30sec or so time frame, that ended up really bad. Even worse, since we used the "only show results after the poll is closed" setting, it meant that at the time the poll was closed all connected clients fired all of these requests at the same time. We ended up disabling the User Status app and that significantly improved the situation.

Unfortunately, the load was still too high. One key thing I found also related to the participants list: It seems like none of the avatar PNGs were properly cached, instead taking up to a few seconds to be generated. I don't think this is an issue related to this app, but rather another part of the Nextcloud setup or our hosting provider. In the end, we decided not to show any detailed results at all, which meant that only the poll creator could see the individual voting behavior and thus only one client triggered the PNG generation. It kinda defeated the purpose why we chose the Polls app in the first place, but it enabled us to conclude the GA successfully.

Long story short, I guess I wouldn't try using the Polls app in such a situation again (especially since there's also specialized software like [OpenSlides](https://github.com/OpenSlides/OpenSlides)), but as long as the load doesn't get too heavy, it actually works surprisingly well.

**Comment by dartcafe** [comment]
Thanks for sharing your experience. I think the user status is something, what needs improvement, but has to be done in the core.

What I am curious about is, what happens in this situation with the current version (1.8 or up), because we changed from frequent polling for updates to a long poll, which reacts to changes in the database. 

In this situation a number of long polls would stay open and result in a frequent update of the store, because every change from any participant gets updated in every participants client. If you have the chance to test this, I would be really interested, if this change would worsen or improve the situation.

**Comment by burnoutberni** [comment]
Ah yes, sounds intriguing. Unfortunately, I'm not in a situation to test this anymore.
